conan/api/conan_api.py

```diff
@@ -60,6 +57,18 @@ def __init__(self, cache_folder=None):
         self.lockfile = LockfileAPI(self)
         self.local = LocalAPI(self)
 
-        required_range_new = self.config.global_conf.get("core:required_conan_version")
-        if required_range_new:
-            validate_conan_version(required_range_new)
+        _check_conan_version(self)
+
+    def reinit(self):
+        self.config.reinit()
+        self.config.migrate()
```

Comment:
Should this migrate?

Reply:
Probably not, forcing a new migration should probably be explicit too.